src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -109,7 +109,30 @@ std::shared_ptr<MulticlassLabels> CARTree::apply_multiclass(std::shared_ptr<Feat
 	auto current=get_root()->as<bnode_t>();
 
 	require(current, "Tree machine not yet trained.");
-	return apply_from_current_node(data->as<DenseFeatures<float64_t>>(), current)->as<MulticlassLabels>();
+	if (auto tmp =
+	        std::dynamic_pointer_cast<DenseSubSamplesFeatures<float64_t>>(data))
+	{
+		auto num_feat = tmp->get_dim_feature_space();
+		auto num_vec = tmp->get_num_vectors();
+		shogun::SGMatrix<float64_t> feature_matrix;
```

Comment:
You don't need to add the namespace shogun here.
Doesn't this cause a segfault? You are allocating an empty matrix and then below you access the memory directly (`feature_matrix.matrix`)?

Reply:
i will repair this as `SGMatrix<float64_t> feature_matrix(num_feat, num_vec);`